### PR TITLE
[cmake] ceres specify SuiteSparse as required

### PR DESCRIPTION
**Pull request by fabiencastan**

### CODE REVIEW

**Comment by fabiencastan** [comment]
Need to add it on travis.

**Comment by fabiencastan** [comment]
Now rebased on popart_develop.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -272,7 +272,8 @@ ADD_DEFINITIONS(-DEIGEN_MPL2_ONLY)
 # - external if CERES_DIR_HINTS and FIND_PACKAGE return a valid Ceres setup
 # ==============================================================================
 MESSAGE(STATUS "Trying to find package Ceres for openMVG: ${CERES_DIR_HINTS}")
-FIND_PACKAGE(Ceres QUIET HINTS ${CERES_DIR_HINTS})
+FIND_PACKAGE(Ceres QUIET HINTS ${CERES_DIR_HINTS}
+                   REQUIRED SuiteSparse)
```
Comment (simogasp) [inline src/CMakeLists.txt]: it does not work for me. `REQUIRED` may  just overrule `QUITE` but it does not means that it only works if ceres is compiled with SuiteSparse. Indeed I tested on a ceres version without SS and it found correctly ceres without complaining.

**Comment by simogasp** [comment]
also, while at it, even if is a bit out of scope, we can remove the FindCeres.cmake in src/cmakemodules, as it was previously used by cctag and it is not the case anymore (and also is quite wrong)

File: src/CMakeLists.txt (hunk 1 above)
Comment (simogasp) [inline src/CMakeLists.txt]: ah ok sorry, my ceres is not up-to-date. Anyway I think the correct way to do that is to use `COMPONENTS` as explained here https://github.com/ceres-solver/ceres-solver/blame/master/cmake/CeresConfig.cmake.in#L318

Another thing is that we may want to allow people to use ceres without SS? Maybe we can put this `find_package` in a if:

```cmake
set(OPENMVG_REQUIRE_CERES_WITH_SUITESPARSE ON)
...
if(OPENMVG_REQUIRE_CERES_WITH_SUITESPARSE)
  FIND_PACKAGE(Ceres REQUIRED HINTS ${CERES_DIR_HINTS} COMPONENTS
else()
 FIND_PACKAGE(Ceres QUIET HINTS ${CERES_DIR_HINTS}
endif()
```

Comment (fabiencastan) [inline src/CMakeLists.txt]: yes, good idea and maybe with a message like:
```
set(OPENMVG_REQUIRE_CERES_WITH_SUITESPARSE ON)
...
if(OPENMVG_REQUIRE_CERES_WITH_SUITESPARSE)
  MESSAGE(INFO "By default, Ceres required SuiteSparse to ensure best performances. If you explicitly need to build without it, you can use the option: -DOPENMVG_REQUIRE_CERES_WITH_SUITESPARSE=OFF")
  FIND_PACKAGE(Ceres REQUIRED HINTS ${CERES_DIR_HINTS} COMPONENTS ...
else()
 FIND_PACKAGE(Ceres QUIET HINTS ${CERES_DIR_HINTS}
endif()
```

**Comment by fabiencastan** [comment]
@simogasp: I don't understand you remark regarding `FindCeres.cmake`. It has been removed in cctag repo, but here it's needed.

**Comment by simogasp** [comment]
nope, if you look at the history it has been added with `FindCCTag` by Lilian. When we build ceres we don't use the FindCeres, we give the path of the `CeresConfig.cmake` (which is better as it contains the definitions used for compiling the lib etc)

**Pull request merged**